worlds/tloz/Rules.py

```diff
@@ -0,0 +1,145 @@
+            add_rule(world.get_location(location.name, player),
+                     lambda state: state.has("Stepladder", player))
+
+    add_rule(world.get_location("Level 5 Boss", player),
+             lambda state: state.has("Recorder", player))
+
+    add_rule(world.get_location("Level 6 Boss", player),
+             lambda state: state.has("Bow", player) and state.has_group("arrows", player))
+
+    add_rule(world.get_location("Level 7 Item (Red Candle)", player),
+             lambda state: state.has("Recorder", player))
+    add_rule(world.get_location("Level 7 Boss", player),
+             lambda state: state.has("Recorder", player))
+    if world.ExpandedPool[player]:
+        add_rule(world.get_location("Level 7 Key Drop (Stalfos)", player),
+                 lambda state: state.has("Recorder", player))
+        add_rule(world.get_location("Level 7 Bomb Drop (Digdogger)", player),
+                 lambda state: state.has("Recorder", player))
+        add_rule(world.get_location("Level 7 Rupee Drop (Dodongos)", player),
+                 lambda state: state.has("Recorder", player))
+
+    for location in food_locations:
+        if world.ExpandedPool[player] or "Drop" not in location:
+            add_rule(world.get_location(location, player),
+                     lambda state: state.has("Food", player))
+
+    add_rule(world.get_location("Level 8 Item (Magical Key)", player),
+             lambda state: state.has("Bow", player) and state.has_group("arrows", player))
+    if world.ExpandedPool[player]:
+        add_rule(world.get_location("Level 8 Bomb Drop (Darknuts North)", player),
+                 lambda state: state.has("Bow", player) and state.has_group("arrows", player))
+
+    for location in tloz_world.levels[9].locations:
+        add_rule(world.get_location(location.name, player),
+                 lambda state: state.has("Triforce Fragment", player, 8) and
+                               state.has_group("swords", player))
+
+    # Yes we are looping this range again for Triforce locations. No I can't add it to the boss event loop
+    for level in range(1, 9):
+        add_rule(world.get_location(f"Level {level} Triforce", player),
+                 lambda state, l=level: state.has(f"Boss {l} Defeated", player))
+
+    # Sword, raft, and ladder spots
+    add_rule(world.get_location("White Sword Pond", player),
+             lambda state: state.has("Heart Container", player, 2))
+    add_rule(world.get_location("Magical Sword Grave", player),
+             lambda state: state.has("Heart Container", player, 9))
+
+    stepladder_locations = [
+        "Ocean Heart Container", "Level 4 Triforce", "Level 4 Boss",
+        "Level 4 Map", "Level 4 Key Drop (Keese North)"
+    ]
+    for location in stepladder_locations:
+        add_rule(world.get_location("Ocean Heart Container", player),
+                 lambda state: state.has("Stepladder", player))
+
+    if world.StartingPosition[player] != 2:
```

Comment:
I'd recommend importing the actual Option class and using the value, or using a string comparison, instead of a hardcoded number comparison that doesn't mean anything when you reread this code in 3 months.